components/form-builder/app/responses/Dialogs/DownloadDialog.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect } from "react";
+import { Dialog, useDialogRef } from "../../shared";
+import { useTranslation } from "react-i18next";
+import { Button } from "@components/globals";
+import { logMessage } from "@lib/logger";
+import axios from "axios";
+import { DownloadFormat } from "@lib/responseDownloadFormats/types";
+import JSZip from "jszip";
+
+export const DownloadDialog = ({
+  checkedItems,
+  isVisible,
+  setIsVisible,
+  downloadError,
+  setDownloadError,
+  formId,
+  onSuccessfulDownload,
+}: {
+  checkedItems: Map<string, boolean>;
+  isVisible: boolean;
+  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  downloadError: boolean;
+  setDownloadError: React.Dispatch<React.SetStateAction<boolean>>;
+  formId: string;
+  onSuccessfulDownload: () => void;
+}) => {
+  const dialogRef = useDialogRef();
+  const { t } = useTranslation("form-builder-responses");
+  const [selectedFormat, setSelectedFormat] = React.useState<DownloadFormat>();
+  const [zip, setZip] = React.useState<boolean>(true);
+
+  useEffect(() => {
+    if (selectedFormat === DownloadFormat.HTML_ZIPPED) {
+      setZip(true);
+    }
+  }, [selectedFormat]);
+
+  if (downloadError) {
+    setDownloadError(false);
+  }
+
+  const handleClose = () => {
+    setSelectedFormat(undefined);
+    setZip(true);
+    setIsVisible(false);
+    dialogRef.current?.close();
+  };
+
+  const downloadFileFromBlob = (data: Blob, fileName: string) => {
+    const href = window.URL.createObjectURL(data);
+    const anchorElement = document.createElement("a");
+    anchorElement.href = href;
+    anchorElement.download = fileName;
+    document.body.appendChild(anchorElement);
+    anchorElement.click();
+    document.body.removeChild(anchorElement);
+    window.URL.revokeObjectURL(href);
+  };
+
+  const handleDownload = async () => {
+    const url = `/api/id/${formId}/submission/download?format=${selectedFormat}`;
+    const ids = Array.from(checkedItems.keys());
+
+    try {
+      if (selectedFormat === DownloadFormat.HTML_ZIPPED) {
+        const response = await axios({
+          url,
+          method: "POST",
+          responseType: "blob",
+          data: {
+            ids: ids.join(","),
+          },
+        });
+
+        const fileName = `records.zip`;
```

Comment:
Maybe add a date to this so it's unique

Reply:
Going to work on filenames with Carolyn today

Reply:
Will add in separate PR